disnake/guild_scheduled_event.py

```diff
@@ -261,16 +261,77 @@ async def delete(self) -> None:
         """
         await self._state.http.delete_guild_scheduled_event(self.guild_id, self.id)
 
+    # note: unlike `Guild.create_scheduled_event`, we don't remove `entity_type`-specific defaults
+    # from parameters in the overloads here, as the entity_type might be the same as before,
+    # in which case those parameters don't have to be passed
+
+    # no entity_type specified
+    @overload
+    async def edit(
+        self,
+        *,
+        name: str = ...,
+        description: Optional[str] = ...,
+        image: Optional[AssetBytes] = ...,
+        channel: Optional[Snowflake] = ...,
+        privacy_level: GuildScheduledEventPrivacyLevel = ...,
+        scheduled_start_time: datetime = ...,
+        scheduled_end_time: Optional[datetime] = ...,
+        entity_metadata: Optional[GuildScheduledEventMetadata] = ...,
+        status: GuildScheduledEventStatus = ...,
+        reason: Optional[str] = ...,
+    ) -> GuildScheduledEvent:
+        ...
+
+    # new entity_type is `external`, no channel
+    @overload
+    async def edit(
+        self,
+        *,
+        entity_type: Literal[GuildScheduledEventEntityType.external],
+        name: str = ...,
+        description: Optional[str] = ...,
+        image: Optional[AssetBytes] = ...,
+        privacy_level: GuildScheduledEventPrivacyLevel = ...,
+        scheduled_start_time: datetime = ...,
+        scheduled_end_time: Optional[datetime] = ...,
```

Comment:
is this supposed to be optional for external events?

Reply:
Nope, good catch. Fixed in 9f9b48793e089dcd5972a2378503c5a708289407